src/pybind/mgr/dashboard/controllers/_base_controller.py

```diff
@@ -265,21 +264,18 @@ def inner(*args, **kwargs):
 
             else:
                 ret = func(*args, **kwargs)
-            if isinstance(ret, bytes):
-                ret = ret.decode('utf-8')
+
+            ret = ret.decode('utf-8') if isinstance(ret, bytes) else ret
+
+            def get_content_type_header(content_type, version):
+                return (f"application/vnd.ceph.api.v{version}+{content_type}"
+                        if version else f"application/{content_type}")
```

Comment:
We [already have a method](https://github.com/VallariAg/ceph/blob/343b580be12d760cbd258da6f02689552a68180f/src/pybind/mgr/dashboard/controllers/_version.py#L50-L55) very similar to this one.

Reply:
Oh, awesome, I'll import that!